lib/pure/json.nim

```diff
@@ -181,6 +181,22 @@ type
     of JArray:
       elems*: seq[JsonNode]
 
+template verifyJsonExists(node: JsonNode, ast: string) =
+  if node == nil:
+    let msg = "Incorrect JSON input. Wanted a JsonNode in $1 but got nil." %
+      [ ast ]
+    raise newException(JsonKindError, msg)
```

Comment:
I was gonna combine these, but I couldn't come up with a clean and inexpensive way to do that.